### PR TITLE
added trustvaluecontroller tests and changed trustvalue type to double

### PR DESCRIPTION
**Pull request by humanjas**
#130 

# Introduction
The current trust value implementation is pretty simple. So it is changed to a way now that the trust value is updated in a decreasing ascending way. Next to that, the trust value type has been changed to a double. All functions, which influence the trust value have been extracted to a new controller: TrustController

# Purpose
This way, the trust value will be calculated more accurately and the trust value will reach the upper limit slower. And because the TrustController has been extracted, the code has increased in readability and maintainability.

# Future Applicability
Use the TrustController to update the trust values. The enums 'FAILED_TRANSACTION', 'SUCCESFUL_TRANSACTION' and 'VERIFIED_IBAN' have to be removed later, after the refactor

# PR Reflection
-

### CODE REVIEW

File: app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
     /**
      * Predefined trust values
      */
-    INITIALIZED(20), REVOKED(0), VERIFIED(50), SUCCESFUL_TRANSACTION(10), FAILED_TRANSACTION(-10);
+    INITIALIZED(10), REVOKED(0), VERIFIED(20), SUCCESFUL_TRANSACTION(10), FAILED_TRANSACTION(-10);
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java]: What is your motivation for choosing these specific values?

File: app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ public View getView(final int position, View convertView, ViewGroup parent) {
             ibanItemText.setText(blockController.getBlocks(ownerName).get(position).getOwner().getIban());
             ImageView pic = (ImageView) view.findViewById(R.id.trust_image);
             pic.setImageResource(
-                    getImageNo(blockController.getBlocks(ownerName).get(position).getTrustValue()));
+                    getImageNo((int) Math.ceil(blockController.getBlocks(ownerName).get(position).getTrustValue())));
```
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: I think Math.ceil isn't doing what you want. Math.ceil always rounds the number upwards, I guess you want to use Math.round.

Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: Will change it

File: app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java (hunk 1 above)
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java]: The initialized value is a value which has to be a bit above 0, but not too much and since the upper limit is 100, it will reach it far too fast. The values of succesful transaction ,failed transaction, verified iban have to be removed later

Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java]: Why not remove it right now?

Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java]: Because the BlockController will be refactored later, in another pull request. When the blockcontroller will be refactored, the old trust value functions will be removed and so also these dependencies. When the dependencies get removed, these values will be removed too.

File: app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * failedTransaction
+     * Calculates the new trust value given a failed transaction
+     *
+     * @param block the given block
+     * @return the block with the new trust value
+     */
+    Block failedTransaction(Block block) {
+        final double newValue = checkCeiling(distributionFunction(getX(block.getTrustValue()) -
```
Comment (nzarin) [inline app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java]: since you only check whether the new value is under 0, does this mean that your max trust can be more than 100?

File: app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java [hunk 4]
```diff
@@ -45,7 +45,7 @@ public static final Block createBlock(BlockData data) throws HashException {
     @Deprecated
     public static final Block getBlock(String type, String newOwner, int sequenceIndex, String hash,
```
Comment (nzarin) [inline app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java]: will this deprecated method be in the master ? Or is this just a temporary solution.

File: app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java (hunk 3 above)
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java]: The distribution function ensures that the value will not go over 100

File: app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java (hunk 4 above)
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java]: Will be refactored in the another pull request.

Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java]: Next to that, this pull request will merge in the develop.

File: app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java (hunk 3 above)
Comment (nzarin) [inline app/src/main/java/nl/tudelft/b_b_w/controller/TrustController.java]: okay, check 👍

File: app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java (hunk 1 above)
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/model/TrustValues.java]: Fair enough

File: app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java (hunk 4 above)
Comment (nzarin) [inline app/src/main/java/nl/tudelft/b_b_w/model/block/BlockFactory.java]: Yeah I understand that it will be merged into develop. But I didn't know you would refactor it in another PR. Clear now. Will approve

**Comment by nzarin** [comment]
Also something for in the 'future'. We might wanna take a look at PageRank (google's algorithm. It is also used for tribler). For now this will work.

**Comment by humanjas** [comment]
Like the other pull request, 3/4 travis builds run and passed. The other timed out, because the android emulator did not run. I have already run the job again several times, but it did timed out again. The code has no dependencies on the newer sdk, so it is a problem of travis.
